nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/test/groovy/org/apache/nifi/lookup/maxmind/TestIPLookupService.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.lookup.maxmind;
+
+import com.maxmind.geoip2.DatabaseReader;
+import org.apache.nifi.lookup.TestProcessor;
+import org.apache.nifi.serialization.record.Record;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+
+public class TestIPLookupService {
+    private TestRunner runner;
+    private IPLookupService testSubject;
+
+    private DatabaseReader mockDatabaseReader;
+
+    @BeforeEach
+    public void setUp() {
+        runner = TestRunners.newTestRunner(TestProcessor.class);
+        testSubject = new IPLookupService() {
+            @NotNull
+            @Override
+            DatabaseReader createDatabaseReader(File dbFile) throws IOException {
+                return mockDatabaseReader;
+            }
+        };
+
+        mockDatabaseReader = Mockito.mock(DatabaseReader.class);
+    }
+
+    @Test
+    void testLookupDefaultNoResult() throws Exception {
+        // GIVEN
+        runner.addControllerService("testSubject", testSubject);
+        runner.setProperty(testSubject, IPLookupService.GEO_DATABASE_FILE, "src/test/resources/dummy.mmdb");
```

Comment:
Instead of checking in `dummy.mmdb` as an empty file, using the `@TempDir` annotation and writing out an empty file before all test methods would avoid the need to commit extra files.

Reply:
Not a fan but fine. (Temp files in tests tend to get messy because developers debug and terminate them all the time without letting the cleanup properly occurring.)